core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/PojoMapper.java

```diff
@@ -95,6 +96,12 @@ public T map(ResultSet rs, StatementContext ctx) throws SQLException {
 
         for (PojoProperty<T> property : properties.getProperties().values()) {
             Nested anno = property.getAnnotation(Nested.class).orElse(null);
+            JdbiProperty propertyAnno = property.getAnnotation(JdbiProperty.class).orElse(JdbiProperty.DEFAULT);
+
+            if (!propertyAnno.mappable()) {
+                unmatchedColumns.remove(property.getName());
```

Comment:
`property.getName()` is a bean property name, but `unmatchedColumn` contains column names. So a column name like `foo_bar` wouldn't be matched up to a `fooBar` Java property.

This check probably needs to be farther down, with the `findColumnIndex()` stuff.